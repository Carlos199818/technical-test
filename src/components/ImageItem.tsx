import { Box, Button, Grid, Typography } from "@mui/material";
import type { RoverPhoto } from "../interfaces/RoverPhotoInterface";


export default function ImageItem({ item, onDelete }: { item: RoverPhoto; onDelete: (imageToDelete: RoverPhoto) => void }) {


    return (
        <Grid size={4} key={item.id}>
            <Box
                sx={{
                    width: "100%",
                    height: "480px",
                    backgroundColor: "#2b2b2b",
                }}>
                <Box sx={{
                    height: "10%",
                }}>
                    <Typography variant="h6" sx={{
                        color: "#fff",
                        textAlign: "center",
                        fontWeight: "bold",
                        paddingTop: "10px",
                    }}>
                        {item.camera.full_name}
                    </Typography>
                </Box>
                <img
                    src={item.img_src}
                    alt="Mars"
                    style={{
                        width: "100%",
                        height: "80%",
                        objectFit: "cover",
                        borderRadius: "10px",
                    }}
                />
                <Box sx={{
                    height: "10%",
                }}>
                    <Button onClick={() => onDelete(item)} sx={{
                        backgroundColor: "#2b2b2b",
                        color: "#f00",
                        "&:hover": {
                            backgroundColor: "#000000",
                        },
                    }}>
                        Eliminar
                    </Button>
                </Box>
            </Box>

        </Grid>
    )
}