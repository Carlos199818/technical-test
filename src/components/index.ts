

export { default as Navbar } from "./Navbar"
export { default as BoxImages } from "./BoxImages"
export { default as ImageItem } from "./ImageItem"