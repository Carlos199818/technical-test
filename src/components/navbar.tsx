import { AppBar, Grid, Typography } from "@mui/material"


export default function Navbar({ navHeight }: { navHeight: number }) {

    return (
        <AppBar
            sx={{
                backgroundColor: "#2b2b2b",
                height: `${navHeight}px`,
            }}
        >
            <Grid container sx={{ p: 5 }}>

                <Grid size={2}>
                    <Typography>
                        Bienvenidos a Marte
                    </Typography>
                </Grid>
                <Grid size={8} sx={{ textAlign: "center" }}>
                    <Typography variant="h4">
                        Ven y disfruta de la experiencia de conocer el planeta rojo desde la comodidad de tu hogar
                    </Typography>
                </Grid>
                <Grid size={2} sx={{ textAlign: "right" }}>
                    <Typography>
                        Welcome to Mars
                    </Typography>
                </Grid>

            </Grid>

        </AppBar>
    )
} 