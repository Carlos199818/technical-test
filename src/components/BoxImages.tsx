import { Box, Button, Grid, Typography } from "@mui/material";
import { useEffect, useState } from "react";
import type { RoverPhoto } from "../interfaces/RoverPhotoInterface";
import axiosInstance from "../api/client";
import ImageItem from "./ImageItem";

export default function BoxImages() {
    const itemsPerPage: number = 12;

    const [data, setData] = useState<RoverPhoto[]>([]);
    const [visibleCount, setVisibleCount] = useState<number>(itemsPerPage);

    const handleShowMore = (): void => {
        setVisibleCount((prev: number) => prev + itemsPerPage);
    };

    const handleDelete = (imageToDelete: RoverPhoto): void => {
        setData((prevItems) => prevItems.filter((image) => image !== imageToDelete));
    };

    useEffect(() => {
        axiosInstance("/photos?sol=1000&api_key=DEMO_KEY")
            .then((response) => {
                setData(response.data.photos);
            })
            .catch((error) => {
                console.error("Error al obtener los datos:", error);
            });
    }, []);


    return (
        <Grid container spacing={5} sx={{ p: 5 }}>

            {
                data.slice(0, visibleCount).map((item: RoverPhoto) => (
                    <ImageItem
                        key={item.id}
                        item={item}
                        onDelete={handleDelete} 
                    />
                ))
            }


            <Grid size={12}>
                <Box
                    sx={{
                        width: "100%",
                        height: "70px",
                        display: "flex",
                        justifyContent: "center",
                    }}>
                    <Button variant="contained" onClick={handleShowMore} sx={{
                        backgroundColor: "#2b2b2b",
                        color: "#fff",
                        "&:hover": {
                            backgroundColor: "#000000",
                        },
                    }}>
                        <Typography variant="h6">
                            Más imágenes
                        </Typography>
                    </Button>
                </Box>
            </Grid>
        </Grid>
    )
}