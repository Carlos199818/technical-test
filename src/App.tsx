// import reactLogo from './assets/react.svg'

import { Box } from "@mui/material"
import { BoxImages, Navbar } from "./components"

function App() {

  const navHeight = 240;

  return (
    <Box sx={{
      display: "flex",
      height: "100vh",

    }}>

      <Box>
        <Navbar navHeight={navHeight} />
      </Box>

      <Box
        sx={{
          width: "100%",
          mt: `${navHeight}px`,
        }}
      >
        <BoxImages />
      </Box>
    </Box>
  )
}

export default App
