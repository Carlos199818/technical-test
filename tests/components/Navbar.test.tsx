import { render } from '@testing-library/react';
import { Navbar } from '../../src/components';

describe('Pruebas en <Navbar />', () => {

    const navHeight = 240;

    test('Debe hacer match con el snapshot', () => {

        const { container } = render( <Navbar navHeight={navHeight} /> );
        expect(container).toMatchSnapshot();
        
    });

});