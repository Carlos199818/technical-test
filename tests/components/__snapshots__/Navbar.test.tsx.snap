// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <Navbar /> Debe hacer match con el snapshot 1`] = `
<div>
  <header
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-1rp7yau-MuiPaper-root-MuiAppBar-root"
    style="--Paper-shadow: 0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12);"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-row css-6r5ftf-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-direction-xs-row MuiGrid-grid-xs-2 css-15bfe7d-MuiGrid-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
        >
          Bienvenidos a Marte
        </p>
      </div>
      <div
        class="MuiGrid-root MuiGrid-direction-xs-row MuiGrid-grid-xs-8 css-17c5yw0-MuiGrid-root"
      >
        <h4
          class="MuiTypography-root MuiTypography-h4 css-f84ov1-MuiTypography-root"
        >
          Ven y disfruta de la experiencia de conocer el planeta rojo desde la comodidad de tu hogar
        </h4>
      </div>
      <div
        class="MuiGrid-root MuiGrid-direction-xs-row MuiGrid-grid-xs-2 css-jxgdt2-MuiGrid-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
        >
          Bienvenidos a Marte
        </p>
      </div>
    </div>
  </header>
</div>
`;
